lib/src/values/simple_config_list.cc

```diff
@@ -99,7 +99,9 @@ namespace hocon {
             }
 
             if (equal(begin(), end(), o.begin(),
-                      [](shared_value const& a, shared_value const& b) { return a == b; })) {
+                      [](shared_value const& a, shared_value const& b) {
```

Comment:
I think you misunderstood the function here. It checks if the pointers are equal, then it does a deep comparison below if that failed.